fishing.dm

```diff
@@ -0,0 +1,287 @@
+		for(var/i=1, i<=carpnumber, i++)
+			var/atom/movable/x = new carp_type
+			x.loc = T
+			if(prob(50))
+				for(var/j = 1, j <= rand(1, 3), j++)
+					step(x, pick(NORTH,SOUTH,EAST,WEST))
+
+
+
+
+
+
+/obj/item/weapon/fishing/pole/afterattack(atom/I, mob/user, proximity)
+	if(istype(I, /turf/open/fishing/water))
+		baited = bait_power * chance //Bait power multiplies the rod's power
+		user.visible_message("<span class='warning'>[user] casts their fishing line into the [I].</span>")
+		if(do_after(user, fishingspeed))
+			if(prob(70)) //change this later
+				user.visible_message("<span class='warning'>A wild carp bites the line and eats the bait! PULL!!!!.</span>")
+				rollthedice()
+			else //100 minus the probability gives us the remaining probability that it fucks up
+				user.visible_message("<span class='warning'>Your line dangles there akwardly as no fish take the bait.</span>")
+				return
+
+
+
+
+
+/obj/item/weapon/fishing/bait
+	name = "Fish Bait"
+	desc = "A ball of stuff that's meant to be bait, what else"
+	icon = 'icons/obj/fishing.dmi'
+	icon_state = "bait" //remember to change this
+	var/bait_power = 1.3 //basic bait only gives a teeny boost
+	var/cancatch = 1 //Only able to catch basic carp with this crappy bait
+
+/obj/item/weapon/fishing/bait/metal
+	name = "Metal Bait"
+	desc = "A ball of metal formed in a carp edible shape."
+	icon = 'icons/obj/fishing.dmi'
+	icon_state = "bait_metal" //remember to change this
+	bait_power = 1.5 //gives it a good kick, but not masses
+	cancatch = 2 //Basic carp and radioactive carp can be caught
+
+/obj/item/weapon/fishing/bait/glowing
+	name = "Glowing Bait"
+	desc = "A ball of luminescant bait."
+	icon = 'icons/obj/fishing.dmi'
+	icon_state = "bait_glowing" //remember to change this
+	bait_power = 1.7 //higher tier bait
+	cancatch = 3 //deep carp can be caught
+
+
+
+//ree
+
+
+
+/////Pulsermajig/////
+
+	//, its depth is set to [depth] and its power is at setting [setting], CTRL click it to set its intensity, and click it normally to set depth//
+
+/obj/item/weapon/fishing/pulser
+	name = "Sonic Pulser"
+	desc = "This nifty device is used to draw lurking carp from the depths"
+	icon = 'icons/obj/fishing.dmi'
+	icon_state = "pulser"
+	var/depth = 0 //depth is set manually
+	var/setting = 0 //How powerful is the pulse, depth dictates what carp, this dictates how MANY
+
+/obj/item/weapon/fishing/pulser/attack_self(mob/user)
+	var/mode = input("Sonic Pulser.", "Modify Settings", "cancel") in list("Modify Settings", "cancel") //input defined here, may be a bad idea
+	var/depthlevel = input("Depth setting.", "Set Depth Level Modifier", "cancel") in list("1", "2", "3", "4", "5", "cancel")
+	var/pulselevel = input("Pulse setting.", "Set Pulse Intensity", "cancel") in list("1", "2", "3", "4", "5", "cancel")
+	switch(mode)
+		if("Modify Settings")
+			switch(depthlevel) 		//long drawn way of doing it but i'm bad at this soooo.
+				if("1")
+					depth = 1
+					icon_state = "pulser_1"
```

Comment:
You define depth and icon_state twice here for no real reason. Also use 
`depth = text2num(depthlevel)
icon_state = "pulser_[depthlevel]"`